docs/proposals/proposal-bug-1512042.md

```diff
@@ -0,0 +1,80 @@
+# Proposal for [Bug 1512042](https://bugzilla.redhat.com/show_bug.cgi?id=1512042)
+
+## Introduction
+The issue is that when a custom namespace (not openshift) is used for holding the images the broker does not currently give access to that namespace when creating the service account for the sandbox. This cause's the resulting transient namespace to get stuck in an error pulling the image. 
+
+# Problem Description
+The problem is that we want to be able to control how the privileges get escalated into that namespace to see/pull the images. This means that we will need a robust solution, that is not a simple bug fix.
+
+## Implementation Details
+The proposed changes are adding a configuration option to the local_registry. This configuration option will be similar to the auto_escalate feature we already have. The option will allow for checking if the originating user has access to the custom namespace to pull images. This will be the default, and you can specify to auto grant access to the namespace. There will only be two options for the configuration value `CheckAccessToNamespace` and `AutoGrantAccessToNamespace`.
+
+To grant access to sandbox's service account we will grant access to pull images from the custom namespace. 
+
+There is also a third option that we should document, which is to allow everyone to access the images in that custom namespace. This is purelly an RBAC policy setting and will allow all the users to access the images. 
+
+### Open Questions
+1. I think this check should be in the `pkg/handler/handler.go` file. This is an authorization check in my mind and should be treated as such. This check is only needed when an APB is being launched and should follow the placement of the current role authorization check.
```

Comment:
My auth stuff has a filter that can do this before the actual handler call, not sure if that's too early or not.